google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -105,67 +226,99 @@ private void runExample(GoogleAdsClient googleAdsClient, long managerId) {
-            continue;
-          }
+            if (customerClient.getId().getValue() == customerId) {
+              continue;
+            }
 
-          // For all level-1 (direct child) accounts that are manager accounts, the above
-          // query will be run against them to create a map of managers to their
-          // child accounts for printing the hierarchy afterwards.
-          customerIdsToChildAccounts.put(customerId, customerClient);
-          // Checks is the child account is a manager itself so that it can later be processed
-          // and added to the map if it hasn't been already.
-          if (customerClient.getManager().getValue()) {
-            // A customer can be managed by multiple managers, so to prevent visiting the same
-            // customer multiple times, we need to check if it's already in the map.
-            boolean alreadyVisited =
-                customerIdsToChildAccounts.containsKey(customerClient.getId().getValue());
-            if (!alreadyVisited && customerClient.getLevel().getValue() == 1) {
-              managerAccountsToSearch.add(customerClient.getId().getValue());
+            // For all level-1 (direct child) accounts that are manager accounts, the above
+            // query will be run against them to create a map of managers to their
+            // child accounts for printing the hierarchy afterwards.
+            customerIdsToChildAccounts.put(customerId, customerClient);
+            // Checks is the child account is a manager itself so that it can later be processed
+            // and added to the map if it hasn't been already.
+            if (customerClient.getManager().getValue()) {
+              // A customer can be managed by multiple managers, so to prevent visiting the same
+              // customer multiple times, we need to check if it's already in the map.
+              boolean alreadyVisited =
+                  customerIdsToChildAccounts.containsKey(customerClient.getId().getValue());
+              if (!alreadyVisited && customerClient.getLevel().getValue() == 1) {
+                managerAccountsToSearch.add(customerClient.getId().getValue());
+              }
             }
           }
+        } catch (GoogleAdsException gae) {
+          System.out.printf(
+              "Unable to retrieve hierarchy for customer ID %d: %s%n",
+              customerId, gae.getGoogleAdsFailure().getErrors(0).getMessage());
         }
       }
 
-      if (!(rootCustomerClient == null)) {
-        System.out.printf(
-            "The hierarchy of customer ID %d is printed below.%n",
-            rootCustomerClient.getId().getValue());
-        printAccountHierarchy(rootCustomerClient, customerIdsToChildAccounts, depth);
-      } else {
-        System.out.println(
-            "Customer ID is likely a test account, so its customer client "
-                + "information cannot be retrieved");
+      if (rootCustomerClient == null) {
```

Comment:
Could you also include brief comment here what could cause the root customer client to be null?